posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,412 @@
+    public CloudEvent getProperties() {
+        /* java.util.properties does not have a direct way to obtain a byte[] so store in an intermedietary Map first*/
+        Map properties = System.getProperties();
+        Jsonb jsonb = JsonbBuilder.create();
+        /* convert properties map into a JSON string which can then be converted into a byte[]*/
+        String jsonString = jsonb.toJson(properties);
+        return CloudEventBuilder.v1()
+                .withData(jsonString.getBytes())
+                .withDataContentType("application/json")
+                .withId("properties")
+                .withType("java.properties")
+                .withSource(URI.create("http://system.poperties"))
+                .build();
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Timed(name = "queryPropertiesTime",
+            description = "Time needed to query the JVM system properties")
+    @Counted(absolute = true, description
+            = "Number of times the JVM system properties are queried")
+    public CloudEvent queryProperties(CloudEvent query){
+        Map properties = System.getProperties();
+        HashMap<String,String> props = new HashMap<>((Map<String,String>)properties);
+        HashMap<String,String> qProps = new HashMap<String,String>();
+        Jsonb jsonb = JsonbBuilder.create();
+
+        /* Pull data from the data portion of the CloudEvent - this is in binary format so convert it into a standard String*/
+        CloudEventData data = query.getData();
+        String jsonString = new String(data.toBytes(), StandardCharsets.UTF_8);
+
+        /* Take the Json Array data and use that to pull out the request properties */
+        ArrayList<String> tProps = jsonb.fromJson(jsonString, ArrayList.class);
+        for(String key: tProps){
+            qProps.put(key, props.get(key));
+        }
+
+        /* return a CloudEvent with our queried properties */
+        return CloudEventBuilder.v1()
+                .withData(jsonb.toJson(qProps).getBytes())
+                .withDataContentType("application/json")
+                .withId("properties")
+                .withType("java.properties")
+                .withSource(URI.create("http://system.poperties"))
+                .build();
+    }
+
+}
+```
+
+
+=== Updating your IBM Cloud Code Engine application
+The application can now return and receive CloudEvents.  We're can update our application in ICCE.
+To update your application
+
+1. Rebuild your Liberty application with the CloudEvent changes.
+2. Rebuild your docker container and publish to it ICCR, either by updating the version or leaving it as is if you are using the image hash
+3. https://cloud.ibm.com/docs/codeengine?topic=codeengine-update-app#update-app-crimage-console[Update your ICCE application] to use the new version.
+
+With your application updated you can validate your changes in ICCE by invoking using the previous curl commands.
+
+
+== Additional Resources
```

Comment:
I think you need at least a sentence or two about the results.. something like

> You now have a MicroProfile application on Open Liberty that runs as a serverless function in IBM Code Engine"

or whatever appropriately summarizes the result

Reply:
if it's relevant, you might also refer them back to the ICCE console to see the application scale when curl requests come in